orion-server/src/main/java/com/pinterest/orion/core/automation/operator/kafka/BrokersetTopicOperator.java

```diff
@@ -166,6 +166,14 @@ public void operate(KafkaCluster cluster) throws Exception {
 
       if (idealPartitionCount > actualPartitionCount) {
         // expansion
+        // reject expasion if topic has compact config
+        if (topicAssignment.getConfig() != null &&
+                topicAssignment.getConfig().containsKey("cleanup.policy") &&
+                topicAssignment.getConfig().get("cleanup.policy").equals("compact")) {
```

Comment:
I think this check may need to be more comprehensive. Sometimes `cleanup.policy` has multi-values. For example, `compact,delete`. We would want to avoid the expansion if `compact` is in `cleanup.policy`. Sounds to me like current code checks for when `compact` is the sole cleanup policy.